src/clj/harja/palvelin/palvelut/kulut/kulut.clj

```diff
@@ -267,21 +268,48 @@
     ;; Muutetaan negaatioksi, koska kysymyksen asettelu
     (not valikatselmus-pidetty?)))
 
+(defn- validoi-kulu [db {:keys [erapaiva koontilaskun-kuukausi id kohdistukset]} urakka-id]
+  (let [;; Kaikki kutsuttavat validoinnit throwaavat virheen, jos eivät mene läpi
+        _ (varmista-erapaiva-on-koontilaskun-kuukauden-sisalla db koontilaskun-kuukausi erapaiva urakka-id)
+        vanha-erapaiva (when id (:erapaiva (first (q/hae-kulu db {:id id}))))
+        saako-tallentaa (tarkista-saako-kulua-tallentaa db urakka-id erapaiva vanha-erapaiva)
+        _ (when (not saako-tallentaa)
+            (throw (IllegalArgumentException.
+                     (str "Kulu on tai kohdistuu hoitokaudelle, jonka välikatselmus on jo pidetty. Tallentaminen ei enää onnistu!"))))
+
+        ;; Varmistetaan, että kululle ei ole lisätty tehtäväryhmää, joka ei ole voimassa
+        urakan-tiedot (first (urakka-kyselyt/hae-urakka db {:id urakka-id}))
+        _ (when (nil? urakan-tiedot)
+            (throw (IllegalArgumentException.
+                     (str "Kulu yritetään lisätä urakalle, jota ei ole olemassa!"))))
+        urakan-alkuvuosi (-> urakan-tiedot :alkupvm pvm/vuosi)
+        urakan-loppuvuosi (-> urakan-tiedot :loppupvm pvm/vuosi)
+        kohdistukset-joilla-tehtavaryhma (filter #(not (nil? (:tehtavaryhma %))) kohdistukset)
+        _ (mapv
```

Comment:
Tää näyttää vähän hassulta omaan silmään. Miltä kuulostais jos tää ois `doseq` ja tän `let`in arrayn ulkopuolella?

Reply:
Ei tässä tässä käyttötapauksessa, mutta joskus tarvitaan enemmän tietoja, niin lisäsin tuohon vähän reilummin.

Reply:
Mä en hoksaa, että mitä etuja saadaan siitä, että jätetään se letin ulkopuolelle. Mutta muutin sen doseqiksi.

Reply:
Ei oo mitään etuja. Makuasia vaan!